sys/shell/commands/sc_gnrc_icmpv6_echo.c

```diff
@@ -0,0 +1,432 @@
+}
+
+static int _configure(int argc, char **argv, _ping_data_t *data)
+{
+    char *cmdname = argv[0];
+    int res = 0;
+
+    /* jump over cmdname */
+    if ((--argc) == 0) {
+        /* only command name given */
+        res = 1;
+    }
+    ++argv;
+    /* parse command line arguments */
+    for (; argc > 0; argc--, argv++) {
+        if (argv[0][0] != '-') {
+            data->hostname = *argv;
+#ifdef MODULE_SOCK_DNS
+            if (sock_dns_query(data->hostname, &data->host, AF_INET6) == 0) {
+                continue;
+            }
+#endif
+            data->iface = ipv6_addr_split_iface(data->hostname);
+            if (data->iface < KERNEL_PID_UNDEF) {
+#if GNRC_NETIF_NUMOF == 1
+                gnrc_netif_t *netif = gnrc_netif_iter(NULL);
+                if (netif != NULL) {
+                    data->iface = netif->pid;
+                }
+#endif
+            }
+            if (ipv6_addr_from_str(&data->host, data->hostname) == NULL) {
+                res = 1;
+                break;
+            }
+        }
+        else {
+            switch (argv[0][1]) {
+                case 'c':
+                    if (argc--) {
+                        data->count = atoi(*(++argv));
+                        if (data->count > 0) {
+                            continue;
+                        }
+                    }
+                    /* falls through intentionally */
+                case 'h':
+                    res = 1;
+                    continue;
+                    /* falls through intentionally */
+                case 'i':
+                    if (argc--) {
+                        data->interval = (uint32_t)atoi(*(++argv)) * US_PER_MS;
+                        if (data->interval > 0) {
+                            continue;
+                        }
+                    }
+                    /* falls through intentionally */
+                case 's':
+                    if (argc--) {
+                        data->datalen = atoi(*(++argv));
+                        continue;
+                    }
+                    /* falls through intentionally */
+                case 't':
+                    if (argc--) {
+                        data->hoplimit = atoi(*(++argv));
+                        continue;
+                    }
+                    /* falls through intentionally */
+                case 'W':
+                    if (argc--) {
+                        data->timeout = (uint32_t)atoi(*(++argv)) * US_PER_MS;
+                        if (data->timeout > 0) {
+                            continue;
+                        }
+                    }
+                    /* falls through intentionally */
+                default:
+                    res = 1;
+                    break;
+            }
+        }
+    }
+    if (res != 0) {
+        _usage(cmdname);
+    }
+#ifdef MODULE_LUID
+    luid_custom(&data->id, sizeof(data->id), DEFAULT_ID);
```

Comment:
This gets done even if the command gets called without arguments. Is that a problem?

Reply:
`data` is a on the stack of the calling thread and get's re-initialized every time the shell handler is called, so no harm done (the counter in luid may go up, but so does e.g. the PID counter in UNIX as well).